partiql-eval/src/eval.rs

```diff
@@ -46,56 +44,172 @@ pub enum EvaluationError {
     InvalidEvaluationPlan(String),
 }
 
+pub trait Evaluable: Debug {
+    fn evaluate(&mut self, ctx: &dyn EvalContext) -> Option<Value>;
+    fn update_input(&mut self, input: &Value);
+}
+
 #[derive(Debug)]
-pub struct Scan {
+pub struct EvalScan {
     pub expr: Box<dyn EvalExpr>,
     pub as_key: String,
+    pub at_key: String,
     pub output: Option<Value>,
 }
 
-impl Scan {
-    pub fn new(expr: Box<dyn EvalExpr>, as_key: &str) -> Self {
-        Scan {
+impl EvalScan {
+    pub fn new(expr: Box<dyn EvalExpr>, as_key: &str, at_key: &str) -> Self {
+        EvalScan {
             expr,
             as_key: as_key.to_string(),
+            at_key: at_key.to_string(),
             output: None,
         }
     }
 }
 
-impl DagEvaluable for Scan {
+impl Evaluable for EvalScan {
     fn evaluate(&mut self, ctx: &dyn EvalContext) -> Option<Value> {
-        let mut value = partiql_bag!();
+        let mut value = partiql_bag![];
         let v = self.expr.evaluate(&Tuple(HashMap::new()), ctx);
+        let ordered = &v.is_ordered();
+        let mut c: i64 = 0;
```

Comment:
What does `c` represent? Is it the counter for the `at` index? If so, could have a more descriptive variable name.

Reply:
Yes it is a counter for `at_index`, will make the change.